app.py

```diff
@@ -251,37 +251,49 @@ def rest():
 
 @app.route("/rest/analyze", methods=["POST", "GET"])
 def rest_analyze():
-    status = el.ERR_REQ_UNEXPECTED
+    status = err.ERR_REQ_UNEXPECTED
+    is_parent = False
     rest_result = {}
     ret = {}
     url = ""
+    raw_url = ""
     if request.method == "POST":
         if "Url" not in request.form:
-            status = el.ERR_BAD_REQ
+            status = err.ERR_BAD_REQ
 
             ret["result"] = rest_result
-            ret["msg"] = el.get_error_message(status)
+            ret["msg"] = err.get_error_message(status)
             ret["status"] = status
             return jsonify(ret)
         else:
-            url = request.form["Url"]
+            raw_url = request.form["Url"]
 
     elif request.method == "GET":
         if "Url" not in request.args:
-            status = el.ERR_BAD_REQ
+            status = err.ERR_BAD_REQ
 
             ret["result"] = rest_result
-            ret["msg"] = el.get_error_message(status)
+            ret["msg"] = err.get_error_message(status)
             ret["status"] = status
             return jsonify(ret)
         else:
-            url = request.args.get("Url")
+            raw_url = request.args.get("Url")
+
+    # URL抽出
+    tmp_group = re.search('(?:https?://)?(?P<host>.*?)(?:[:#?/@]|$)', raw_url)
+
+    if tmp_group:
+        host = tmp_group.group('host')
+        if host == "youtube.com" or host == "youtu.be":
```

Comment:
```suggestion
        if host == "youtube.com" or host == "youtu.be":
```
        if host == "www.youtube.com" or host == "youtu.be":

hostにwwwが必要でした。